principles.md

```diff
@@ -0,0 +1,37 @@
+# rextendr design prinicples
+
+This guide documents important internal coding conventions used in rextendr.
+
+## Communicating with the user
+
+rextendr uses the cli package to format messages to the user, which are generally then routed through `rlang::inform()`. Most UI will happen through `ui_*()` functions:
+
+| function | purpose                                                                                                       |
+|----------|---------------------------------------------------------------------------------------------------------------|
+| `ui_v()` | communicate that rextendr has done something, such as write a file                                            |
```

Comment:
I suggest adding "... has done something *successfully*, ..." to emphasize that `*_v` functions communicate positive results.

Reply:
yes, good point -- updated!